core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -76,10 +76,13 @@
     public static final String FETCH_ID_ACTION_NAME = "indices:data/read/search[phase/fetch/id]";
 
     private final TransportService transportService;
+    private final ResponseCollectorService responseCollectorService;
 
-    public SearchTransportService(Settings settings, TransportService transportService) {
+    public SearchTransportService(Settings settings, TransportService transportService,
+                                  ResponseCollectorService responseCollectorService) {
```

Comment:
> I played with this, however, it requires making SearchExecutionStatsCollector public instead of package private, since we need to construct it within the BiFunction passed in to SearchTransportService in Node.

I think that is ok!